### PR TITLE
[WIP] Type hinting

### PR DESCRIPTION
**Pull request by lk-geimfari**
I have implemented type hints for module `utils.py`.

### CODE REVIEW

File: mimesis/utils.py [hunk 1]
```diff
@@ -7,30 +7,35 @@
...
 from random import choice
 
-from mimesis.exceptions import UnexpectedGender, UnsupportedLocale
-from mimesis.settings import SUPPORTED_LOCALES
+from mimesis.exceptions import (
+    UnexpectedGender,
+    UnsupportedLocale,
+)
+import mimesis.settings as settings
+import mimesis.type as types
```
Comment (lk-geimfari) [inline mimesis/utils.py]: I have collected all types in this module and import ones from it. Is it good idea?

File: mimesis/type.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from typing import Dict, List, Union, Mapping, Any
```
Comment (sobolevn) [inline mimesis/type.py]: I would also use the name `typing` for this module. `mimesis.typing`.

File: mimesis/utils.py (hunk 1 above)
Comment (sobolevn) [reply mimesis/utils.py]: Yeap, it is!

**Comment by sobolevn** [comment]
Dropping `python3.3.` and `python3.4` will be a new minor release (or even major!).

**Comment by lk-geimfari** [comment]
@DefaltSimon you should use types from `mimesis.typing`. `utils.py` is great example. Thanks!

**Pull request closed without merging**